tsdb/db.go

```diff
@@ -761,20 +760,24 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 }
 
 func removeBestEffortTmpDirs(l log.Logger, dir string) error {
-	files, err := ioutil.ReadDir(dir)
+	files, err := os.ReadDir(dir)
 	if os.IsNotExist(err) {
 		return nil
 	}
 	if err != nil {
 		return err
 	}
-	for _, fi := range files {
+	for _, f := range files {
+		fi, err := f.Info()
+		if err != nil {
```

Comment:
Shall this error just be ignored then ?

Reply:
I don't think it should be ignore. After all the next call depends on `fi` being there. Is there a problem with returning the error here?

Reply:
The call to Ìnfo()`can be avoid by using directly `fs.DirEntry` instead of `os.FileInfo`. Let's see if it fixes the tests